### PR TITLE
rainbow.cpp: Rainbow 100 A ROM added [bsdimp]

### PR DESCRIPTION
**Pull request by rfka01**

### CODE REVIEW

**Review by cuavas — COMMENTED** [review]
What causes the additional errors?  Is this for the same hardware, or is it for a slightly different revision?

**Comment by rfka01** [comment]
I can't really say, all the Rainbow ROM versions need patches to boot, and the ones we had for the other Rainbow 100 A ROM version probably won't apply for this one. bsdimp grabbed this from a running machine, so I think the ROM is correct.

File: src/mame/drivers/rainbow.cpp [hunk 1]
```diff
@@ -3388,11 +3388,19 @@ void rainbow_modelb_state::rainbow_modelb(machine_config &config)
...
 	ROM_REGION(0x100000, "maincpu", 0)
 
-	ROM_LOAD("23-176e4-00.e89", 0xfa000, 0x2000, CRC(405e9619) SHA1(86604dccea84b46e05d705abeda25b12f7cc8c59)) // ROM (FA000-FBFFF) (E89) 8 K
-	ROM_LOAD("23-177e4-00.e90", 0xfc000, 0x2000, CRC(1ec72a66) SHA1(ed19944ae711e97d6bec34c885be04c4c3c95852)) // ROM (FC000-FDFFF) (E90) 8 K
+	// printer and comm. port error messages, keyboard works, machine boots
+	ROM_SYSTEM_BIOS(0, "040311a", "Version 04.03.11A")
+	ROMX_LOAD("23-176e4-00.e89", 0xfa000, 0x2000, CRC(405e9619) SHA1(86604dccea84b46e05d705abeda25b12f7cc8c59), ROM_BIOS(0)) // ROM (FA000-FBFFF) (E89) 8 K
+	ROMX_LOAD("23-177e4-00.e90", 0xfc000, 0x2000, CRC(1ec72a66) SHA1(ed19944ae711e97d6bec34c885be04c4c3c95852), ROM_BIOS(0)) // ROM (FC000-FDFFF) (E90) 8 K
 	ROM_FILL(0xfa26d, 1, 0x00) // [0xFA000 + 0x026d] disable CRC check   [100-A ROM]
 	ROM_FILL(0xfadea, 1, 0x00) // [0xFA000 + 0x0dea] Floppy workaround: in case of Z80 RESPONSE FAILURE ($80 bit set in AL), don't block floppy access
```
Comment (cuavas) [inline src/mame/drivers/rainbow.cpp:3396]: What are these patches doing to the other BIOS?  Is the code in the same place?

Comment (rfka01) [reply src/mame/drivers/rainbow.cpp:3396]: I guess Bavarese's observations starting at [https://forums.bannister.org/ubbthreads.php?ubb=showflat&Number=79906#Post79906](url) apply, but it's beyond my technical grasp to really follow along.

**Pull request merged**